diff -Nuarp onnxruntime-1.21.0.orig/onnxruntime/core/mlas/inc/mlas.h onnxruntime-1.21.0/onnxruntime/core/mlas/inc/mlas.h
--- onnxruntime-1.21.0.orig/onnxruntime/core/mlas/inc/mlas.h	2025-03-06 08:37:52.000000000 +0800
+++ onnxruntime-1.21.0/onnxruntime/core/mlas/inc/mlas.h	2025-04-01 16:46:01.011675470 +0800
@@ -78,7 +78,7 @@ Abstract:
 //
 
 #if defined(MLAS_TARGET_AMD64) || defined (MLAS_TARGET_POWER)
-#define MLAS_SUPPORTS_GEMM_DOUBLE
+// #define MLAS_SUPPORTS_GEMM_DOUBLE
 #endif
 
 #if (!defined(_MSC_VER)) || (_MSC_VER >= 1930)
@@ -89,7 +89,7 @@ Abstract:
 // When building an universial binary for APPLE, this flag would
 // cause trouble for x64 target.
 
-#define MLAS_F16VEC_INTRINSICS_SUPPORTED
+// #define MLAS_F16VEC_INTRINSICS_SUPPORTED
 
 #endif //
 #endif // ARM64
