diff -Nuarp pytorch-v2.5.0.orig/cmake/Codegen.cmake pytorch-v2.5.0/cmake/Codegen.cmake
--- pytorch-v2.5.0.orig/cmake/Codegen.cmake	2024-10-18 00:29:44.000000000 +0800
+++ pytorch-v2.5.0/cmake/Codegen.cmake	2024-10-18 15:41:38.449256056 +0800
@@ -52,6 +52,11 @@ endif()
 install(FILES ${CMAKE_BINARY_DIR}/caffe2/core/macros.h
         DESTINATION include/caffe2/core)
 
+set(Python3_FIND_STRATEGY "LOCATION" CACHE INTERNAL "" FORCE)
+set(Python3_FIND_REGISTRY "NEVER" CACHE INTERNAL "" FORCE)
+find_package(Python3 COMPONENTS Interpreter)
+set(PYTHON_EXECUTABLE ${Python3_EXECUTABLE})
+
 # ---[ ATen specific
 if(INTERN_BUILD_ATEN_OPS)
   if(MSVC)
