diff -Nuarp pytorch-v2.5.0.orig/cmake/TorchConfig.cmake.in pytorch-v2.5.0/cmake/TorchConfig.cmake.in
--- pytorch-v2.5.0.orig/cmake/TorchConfig.cmake.in	2024-10-18 00:29:44.000000000 +0800
+++ pytorch-v2.5.0/cmake/TorchConfig.cmake.in	2024-10-18 15:36:59.910242775 +0800
@@ -15,7 +15,7 @@
 #   torch
 macro(append_torchlib_if_found)
   foreach (_arg ${ARGN})
-    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib" NO_DEFAULT_PATH)
     if(${_arg}_LIBRARY)
       list(APPEND TORCH_LIBRARIES ${${_arg}_LIBRARY})
     else()
