diff -Nuarp pytorch-v2.7.1.orig/cmake/Codegen.cmake pytorch-v2.7.1/cmake/Codegen.cmake
--- pytorch-v2.7.1.orig/cmake/Codegen.cmake	2025-06-05 02:15:52.000000000 +0800
+++ pytorch-v2.7.1/cmake/Codegen.cmake	2025-07-24 16:52:03.349633860 +0800
@@ -59,6 +59,11 @@ endif()
 install(FILES ${CMAKE_BINARY_DIR}/caffe2/core/macros.h
         DESTINATION include/caffe2/core)
 
+set(Python3_FIND_STRATEGY "LOCATION" CACHE INTERNAL "" FORCE)
+set(Python3_FIND_REGISTRY "NEVER" CACHE INTERNAL "" FORCE)
+find_package(Python3 COMPONENTS Interpreter)
+set(PYTHON_EXECUTABLE ${Python3_EXECUTABLE})
+
 # ---[ ATen specific
 if(INTERN_BUILD_ATEN_OPS)
   if(MSVC)
diff -Nuarp pytorch-v2.7.1.orig/CMakeLists.txt pytorch-v2.7.1/CMakeLists.txt
--- pytorch-v2.7.1.orig/CMakeLists.txt	2025-06-05 02:15:51.000000000 +0800
+++ pytorch-v2.7.1/CMakeLists.txt	2025-07-24 16:52:03.349815810 +0800
@@ -380,15 +380,6 @@ option(USE_MIMALLOC "Use mimalloc" OFF)
 # Enable third party mimalloc library to improve memory allocation performance
 # on Windows.
 option(USE_MIMALLOC_ON_MKL "Use mimalloc on MKL" OFF)
-if(WIN32)
-  set(USE_MIMALLOC ON)
-
-  # Not enable USE_MIMALLOC_ON_MKL due to it caused issue:
-  # https://github.com/pytorch/pytorch/issues/138994
-  # Will turn on when we can fix USE_STATIC_MKL lost functionality:
-  # https://github.com/pytorch/pytorch/pull/138996
-  # set(USE_MIMALLOC_ON_MKL ON)
-endif()
 
 if(USE_CCACHE)
   find_program(CCACHE_PROGRAM ccache)
