diff -Nuarp pytorch-v2.12.1.orig/cmake/TorchConfig.cmake.in pytorch-v2.12.1/cmake/TorchConfig.cmake.in
--- pytorch-v2.12.1.orig/cmake/TorchConfig.cmake.in	2026-07-03 15:12:01.833806514 +0800
+++ pytorch-v2.12.1/cmake/TorchConfig.cmake.in	2026-07-03 15:12:30.210000000 +0800
@@ -15,7 +15,7 @@
 #   torch
 macro(append_torchlib_if_found)
   foreach (_arg ${ARGN})
-    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib" NO_DEFAULT_PATH)
     if(${_arg}_LIBRARY)
       list(APPEND TORCH_LIBRARIES ${${_arg}_LIBRARY})
     else()
@@ -26,7 +26,7 @@ endmacro()
 
 macro(append_wholearchive_lib_if_found)
   foreach (_arg ${ARGN})
-    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib" NO_DEFAULT_PATH)
     if(${_arg}_LIBRARY)
       if(APPLE)
         list(APPEND TORCH_LIBRARIES "-Wl,-force_load,${${_arg}_LIBRARY}")
@@ -121,7 +121,7 @@ endif()
 
 if(@USE_CUDA@)
   if(MSVC)
-    find_library(CAFFE2_NVRTC_LIBRARY caffe2_nvrtc PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(CAFFE2_NVRTC_LIBRARY caffe2_nvrtc PATHS "${TORCH_INSTALL_PREFIX}/lib" NO_DEFAULT_PATH)
     list(APPEND TORCH_CUDA_LIBRARIES ${CAFFE2_NVRTC_LIBRARY})
   else()
     set(TORCH_CUDA_LIBRARIES ${CUDA_NVRTC_LIB})
@@ -131,7 +131,7 @@ if(@USE_CUDA@)
   endif()
 
   if(@BUILD_SHARED_LIBS@)
-    find_library(C10_CUDA_LIBRARY c10_cuda PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(C10_CUDA_LIBRARY c10_cuda PATHS "${TORCH_INSTALL_PREFIX}/lib" NO_DEFAULT_PATH)
     list(APPEND TORCH_CUDA_LIBRARIES ${C10_CUDA_LIBRARY} ${Caffe2_PUBLIC_CUDA_DEPENDENCY_LIBS})
   endif()
   list(APPEND TORCH_LIBRARIES ${TORCH_CUDA_LIBRARIES})
@@ -141,7 +141,7 @@ if(@USE_XPU@ AND @BUILD_SHARED_LIBS@)
     append_torchlib_if_found(c10_xpu torch_xpu)
 endif()
 
-find_library(TORCH_LIBRARY torch PATHS "${TORCH_INSTALL_PREFIX}/lib")
+find_library(TORCH_LIBRARY torch PATHS "${TORCH_INSTALL_PREFIX}/lib" NO_DEFAULT_PATH)
 # the statements below changes target properties on
 # - the imported target from Caffe2Targets.cmake in shared library mode (see the find_package above)
 #    - this is untested whether it is the correct (or desired) methodology in CMake
