diff -Nuarp pytorch-v2.14.0.orig/CMakeLists.txt pytorch-v2.14.0/CMakeLists.txt
--- pytorch-v2.14.0.orig/CMakeLists.txt	2026-09-17 11:04:29.112584117 +0800
+++ pytorch-v2.14.0/CMakeLists.txt	2026-09-17 11:04:29.112632989 +0800
@@ -48,6 +48,8 @@ project(${SKBUILD_PROJECT_NAME}
 
 # When built via scikit-build-core, Python_EXECUTABLE is already set.
 # For standalone cmake invocations, find it via find_package.
+set(Python_FIND_STRATEGY "LOCATION" CACHE INTERNAL "" FORCE)
+set(Python_FIND_REGISTRY "NEVER" CACHE INTERNAL "" FORCE)
 find_package(Python COMPONENTS Interpreter REQUIRED)
 
 # Forward environment variables to CMake variables, replicating the behavior
