diff -Nuarp pytorch-v2.2.1.orig/cmake/TorchConfig.cmake.in pytorch-v2.2.1/cmake/TorchConfig.cmake.in
--- pytorch-v2.2.1.orig/cmake/TorchConfig.cmake.in	2024-02-24 04:00:54.000000000 +0800
+++ pytorch-v2.2.1/cmake/TorchConfig.cmake.in	2024-03-27 19:47:29.315834325 +0800
@@ -15,7 +15,7 @@
 #   torch
 macro(append_torchlib_if_found)
   foreach (_arg ${ARGN})
-    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib" NO_DEFAULT_PATH)
     if(${_arg}_LIBRARY)
       list(APPEND TORCH_LIBRARIES ${${_arg}_LIBRARY})
     else()
