diff -Nuarp pytorch-v2.2.0.orig/CMakeLists.txt pytorch-v2.2.0/CMakeLists.txt
--- pytorch-v2.2.0.orig/CMakeLists.txt	2024-01-31 08:58:01.000000000 +0800
+++ pytorch-v2.2.0/CMakeLists.txt	2024-02-04 11:19:59.509887231 +0800
@@ -351,10 +351,6 @@ cmake_dependent_option(
     BUILD_FUNCTORCH "Build Functorch" ON "BUILD_PYTHON" OFF)
 
 option(USE_MIMALLOC "Use mimalloc" OFF)
-# Enable third party mimalloc library to improve memory allocation performance on Windows.
-if(WIN32)
-  set(USE_MIMALLOC ON)
-endif()
 
 if(USE_CCACHE)
   find_program(CCACHE_PROGRAM ccache)
