diff -Nuarp pytorch-v2.7.1.orig/cmake/TorchConfig.cmake.in pytorch-v2.7.1/cmake/TorchConfig.cmake.in
--- pytorch-v2.7.1.orig/cmake/TorchConfig.cmake.in	2025-06-05 02:15:52.000000000 +0800
+++ pytorch-v2.7.1/cmake/TorchConfig.cmake.in	2025-07-24 16:51:22.740765310 +0800
@@ -15,7 +15,7 @@
 #   torch
 macro(append_torchlib_if_found)
   foreach (_arg ${ARGN})
-    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib" NO_DEFAULT_PATH)
     if(${_arg}_LIBRARY)
       list(APPEND TORCH_LIBRARIES ${${_arg}_LIBRARY})
     else()
