diff -Nuarp pytorch-v1.13.0.orig/cmake/TorchConfig.cmake.in pytorch-v1.13.0/cmake/TorchConfig.cmake.in
--- pytorch-v1.13.0.orig/cmake/TorchConfig.cmake.in	2022-10-29 00:56:05.000000000 +0800
+++ pytorch-v1.13.0/cmake/TorchConfig.cmake.in	2022-11-06 21:24:19.216240317 +0800
@@ -15,7 +15,7 @@
 #   torch
 macro(append_torchlib_if_found)
   foreach (_arg ${ARGN})
-    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib" NO_DEFAULT_PATH)
     if(${_arg}_LIBRARY)
       list(APPEND TORCH_LIBRARIES ${${_arg}_LIBRARY})
     else()
