diff -Nuarp pytorch-v2.12.1.orig/cmake/External/EigenBLAS.cmake pytorch-v2.12.1/cmake/External/EigenBLAS.cmake
--- pytorch-v2.12.1.orig/cmake/External/EigenBLAS.cmake	2026-07-03 15:12:01.845907427 +0800
+++ pytorch-v2.12.1/cmake/External/EigenBLAS.cmake	2026-07-03 15:12:30.211000000 +0800
@@ -38,8 +38,6 @@ set(EigenBlas_SRCS
   ${EIGEN_BLAS_SRC_DIR}/f2c/stbmv.c
   ${EIGEN_BLAS_SRC_DIR}/f2c/ctbmv.c
   ${EIGEN_BLAS_SRC_DIR}/f2c/ztbmv.c
-  ${EIGEN_BLAS_SRC_DIR}/f2c/d_cnjg.c
-  ${EIGEN_BLAS_SRC_DIR}/f2c/r_cnjg.c
   ${EIGEN_BLAS_SRC_DIR}/f2c/complexdots.c
 )
 
