diff -Nuarp pytorch-v2.4.0.orig/CMakeLists.txt pytorch-v2.4.0/CMakeLists.txt
--- pytorch-v2.4.0.orig/CMakeLists.txt	2024-08-08 14:37:20.072270459 +0800
+++ pytorch-v2.4.0/CMakeLists.txt	2024-08-08 14:45:34.552294038 +0800
@@ -384,11 +384,6 @@ cmake_dependent_option(BUILD_BUNDLE_PTXA
                        OFF "USE_CUDA" OFF)
 
 option(USE_MIMALLOC "Use mimalloc" OFF)
-# Enable third party mimalloc library to improve memory allocation performance
-# on Windows.
-if(WIN32)
-  set(USE_MIMALLOC ON)
-endif()
 
 if(USE_CCACHE)
   find_program(CCACHE_PROGRAM ccache)
