diff -Nuarp pytorch-v2.7.0.orig/torch/csrc/inductor/aoti_torch/shim_common.cpp pytorch-v2.7.0/torch/csrc/inductor/aoti_torch/shim_common.cpp
--- pytorch-v2.7.0.orig/torch/csrc/inductor/aoti_torch/shim_common.cpp	2025-04-24 00:19:40.000000000 +0800
+++ pytorch-v2.7.0/torch/csrc/inductor/aoti_torch/shim_common.cpp	2025-04-24 11:16:27.206874692 +0800
@@ -237,7 +237,7 @@ AOTI_TORCH_SCALAR_TO_TENSOR_IMPL(
     ComplexDouble)
 #undef AOTI_TORCH_SCALAR_TO_TENSOR_IMPL
 
-#ifndef C10_MOBILE
+#if 0//ndef C10_MOBILE
 #include <torch/version.h>
 uint64_t aoti_torch_abi_version() {
   return TORCH_ABI_VERSION;
diff -Nuarp pytorch-v2.7.0.orig/torch/csrc/jit/serialization/pickle.cpp pytorch-v2.7.0/torch/csrc/jit/serialization/pickle.cpp
--- pytorch-v2.7.0.orig/torch/csrc/jit/serialization/pickle.cpp	2025-04-24 00:19:40.000000000 +0800
+++ pytorch-v2.7.0/torch/csrc/jit/serialization/pickle.cpp	2025-04-24 11:16:16.386488348 +0800
@@ -66,7 +66,7 @@ std::vector<char> pickle(
 // This has to live here instead of the C++ API to mirror torch.save since the
 // mobile build excludes the C++ API
 std::vector<char> pickle_save(const at::IValue& ivalue) {
-#ifndef C10_MOBILE
+#if 0//ndef C10_MOBILE
   // Pickle the IValue into an array of bytes
   std::vector<char> pickle_data;
   Pickler pickler([&](const char* buf, size_t size) {
@@ -103,7 +103,7 @@ std::vector<char> pickle_save(const at::
 #endif
 }
 
-#ifndef C10_MOBILE
+#if 0//ndef C10_MOBILE
 size_t VectorReader::read(uint64_t pos, void* buf, size_t n, const char* what)
     const {
   std::copy(
@@ -124,7 +124,7 @@ size_t StringViewReader::read(
 
 IValue pickle_load(const std::vector<char>& data) {
   // Read in the pickle data
-#ifndef C10_MOBILE
+#if 0//ndef C10_MOBILE
   caffe2::serialize::PyTorchStreamReader reader(
       std::make_unique<VectorReader>(data));
 
@@ -146,7 +146,7 @@ IValue pickle_load(const std::vector<cha
 
 // A specialized version of pickle_load that can load custom objects.
 c10::IValue pickle_load_obj(std::string_view data) {
-#ifndef C10_MOBILE
+#if 0//ndef C10_MOBILE
   caffe2::serialize::PyTorchStreamReader reader(
       std::make_unique<torch::jit::StringViewReader>(data));
   return torch::jit::readArchiveAndTensors(
diff -Nuarp pytorch-v2.7.0.orig/torch/csrc/jit/serialization/pickle.h pytorch-v2.7.0/torch/csrc/jit/serialization/pickle.h
--- pytorch-v2.7.0.orig/torch/csrc/jit/serialization/pickle.h	2025-04-24 00:19:40.000000000 +0800
+++ pytorch-v2.7.0/torch/csrc/jit/serialization/pickle.h	2025-04-24 11:16:16.386653507 +0800
@@ -106,7 +106,7 @@ TORCH_API IValue unpickle(
     c10::TypePtr (*type_parser)(const std::string&) =
         Unpickler::defaultTypeParser);
 
-#ifndef C10_MOBILE
+#if 0//ndef C10_MOBILE
 class VectorReader : public caffe2::serialize::ReadAdapterInterface {
  public:
   VectorReader(std::vector<char> data) : data_(std::move(data)) {}
