diff --git a/cmake/External/EigenBLAS.cmake b/cmake/External/EigenBLAS.cmake
index 133712c4c1896..1f677f91a6617 100644
--- a/cmake/External/EigenBLAS.cmake
+++ b/cmake/External/EigenBLAS.cmake
@@ -38,8 +38,6 @@ set(EigenBlas_SRCS
   ${EIGEN_BLAS_SRC_DIR}/f2c/stbmv.c
   ${EIGEN_BLAS_SRC_DIR}/f2c/ctbmv.c
   ${EIGEN_BLAS_SRC_DIR}/f2c/ztbmv.c
-  ${EIGEN_BLAS_SRC_DIR}/f2c/d_cnjg.c
-  ${EIGEN_BLAS_SRC_DIR}/f2c/r_cnjg.c
   ${EIGEN_BLAS_SRC_DIR}/f2c/complexdots.c
 )
 
