diff -Nuarp pytorch-v2.9.0.orig/CMakeLists.txt pytorch-v2.9.0/CMakeLists.txt
--- pytorch-v2.9.0.orig/CMakeLists.txt	2025-10-16 01:15:02.000000000 +0800
+++ pytorch-v2.9.0/CMakeLists.txt	2025-10-20 19:42:10.157337219 +0800
@@ -383,15 +383,6 @@ option(USE_MIMALLOC "Use mimalloc" OFF)
 # Enable third party mimalloc library to improve memory allocation performance
 # on Windows.
 option(USE_MIMALLOC_ON_MKL "Use mimalloc on MKL" OFF)
-if(WIN32)
-  set(USE_MIMALLOC ON)
-
-  # Not enable USE_MIMALLOC_ON_MKL due to it caused issue:
-  # https://github.com/pytorch/pytorch/issues/138994
-  # Will turn on when we can fix USE_STATIC_MKL lost functionality:
-  # https://github.com/pytorch/pytorch/pull/138996
-  # set(USE_MIMALLOC_ON_MKL ON)
-endif()
 
 if(USE_CCACHE)
   find_program(CCACHE_PROGRAM ccache)
