diff -Nuarp pytorch-v2.9.0.orig/cmake/Modules/FindARM.cmake pytorch-v2.9.0/cmake/Modules/FindARM.cmake
--- pytorch-v2.9.0.orig/cmake/Modules/FindARM.cmake	2025-10-16 01:15:02.000000000 +0800
+++ pytorch-v2.9.0/cmake/Modules/FindARM.cmake	2025-10-20 19:41:10.579781212 +0800
@@ -152,8 +152,8 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
     ENDMACRO()
 
     # Check for SVE256 vector length
-    CHECK_COMPILES(CXX "SVE256" "-march=armv8.2-a+sve -msve-vector-bits=256" "${SVE_CODE}")
-    CHECK_COMPILES(CXX "ARM_BF16" "-march=armv8.2-a+sve+bf16 -msve-vector-bits=256" "${ARM_BF16_CODE}")
+    # CHECK_COMPILES(CXX "SVE256" "-march=armv8.2-a+sve -msve-vector-bits=256" "${SVE_CODE}")
+    # CHECK_COMPILES(CXX "ARM_BF16" "-march=armv8.2-a+sve+bf16 -msve-vector-bits=256" "${ARM_BF16_CODE}")
 
     # If SVE256 support is not found, set CXX_SVE_FOUND to FALSE and notify the user
     if(NOT CXX_SVE256_FOUND)
