diff -Nuarp pytorch-v2.4.0.orig/torch/csrc/jit/serialization/pickle.cpp pytorch-v2.4.0/torch/csrc/jit/serialization/pickle.cpp
--- pytorch-v2.4.0.orig/torch/csrc/jit/serialization/pickle.cpp	2024-08-08 14:37:20.162270464 +0800
+++ pytorch-v2.4.0/torch/csrc/jit/serialization/pickle.cpp	2024-08-08 14:42:08.266284201 +0800
@@ -37,7 +37,7 @@ std::vector<char> pickle(
 // This has to live here instead of the C++ API to mirror torch.save since the
 // mobile build excludes the C++ API
 std::vector<char> pickle_save(const at::IValue& ivalue) {
-#ifndef C10_MOBILE
+#if 0//ndef C10_MOBILE
   // Pickle the IValue into an array of bytes
   std::vector<char> pickle_data;
   Pickler pickler([&](const char* buf, size_t size) {
@@ -73,7 +73,7 @@ std::vector<char> pickle_save(const at::
 #endif
 }
 
-#ifndef C10_MOBILE
+#if 0//ndef C10_MOBILE
 size_t VectorReader::read(uint64_t pos, void* buf, size_t n, const char* what)
     const {
   std::copy(
@@ -84,7 +84,7 @@ size_t VectorReader::read(uint64_t pos,
 
 IValue pickle_load(const std::vector<char>& data) {
   // Read in the pickle data
-#ifndef C10_MOBILE
+#if 0//ndef C10_MOBILE
   caffe2::serialize::PyTorchStreamReader reader(
       std::make_unique<VectorReader>(data));
 
diff -Nuarp pytorch-v2.4.0.orig/torch/csrc/jit/serialization/pickle.h pytorch-v2.4.0/torch/csrc/jit/serialization/pickle.h
--- pytorch-v2.4.0.orig/torch/csrc/jit/serialization/pickle.h	2024-08-08 14:37:20.162270464 +0800
+++ pytorch-v2.4.0/torch/csrc/jit/serialization/pickle.h	2024-08-08 14:42:00.539283833 +0800
@@ -103,7 +103,7 @@ TORCH_API IValue unpickle(
     c10::TypePtr (*type_parser)(const std::string&) =
         Unpickler::defaultTypeParser);
 
-#ifndef C10_MOBILE
+#if 0//ndef C10_MOBILE
 class VectorReader : public caffe2::serialize::ReadAdapterInterface {
  public:
   VectorReader(std::vector<char> data) : data_(std::move(data)) {}
