diff -Nuarp pytorch-v2.1.2.orig/CMakeLists.txt pytorch-v2.1.2/CMakeLists.txt
--- pytorch-v2.1.2.orig/CMakeLists.txt	2023-12-15 10:03:27.000000000 +0800
+++ pytorch-v2.1.2/CMakeLists.txt	2023-12-15 15:33:50.638638754 +0800
@@ -354,10 +354,6 @@ cmake_dependent_option(
     BUILD_FUNCTORCH "Build Functorch" ON "BUILD_PYTHON" OFF)
 
 option(USE_MIMALLOC "Use mimalloc" OFF)
-# Enable third party mimalloc library to improve memory allocation performance on Windows.
-if(WIN32)
-  set(USE_MIMALLOC ON)
-endif()
 
 if(USE_CCACHE)
   find_program(CCACHE_PROGRAM ccache)
