diff -Nuarp vision-0.21.0.orig/torchvision/csrc/vision.h vision-0.21.0/torchvision/csrc/vision.h
--- vision-0.21.0.orig/torchvision/csrc/vision.h	2025-01-29 09:12:10.000000000 +0800
+++ vision-0.21.0/torchvision/csrc/vision.h	2025-04-01 15:54:15.572769271 +0800
@@ -6,7 +6,4 @@
 namespace vision {
 VISION_API int64_t cuda_version();
 
-namespace detail {
-extern "C" inline auto _register_ops = &cuda_version;
-} // namespace detail
 } // namespace vision
