diff -Nuarp vision-0.22.0.orig/cmake/TorchVisionConfig.cmake.in vision-0.22.0/cmake/TorchVisionConfig.cmake.in
--- vision-0.22.0.orig/cmake/TorchVisionConfig.cmake.in	2025-04-14 16:19:44.000000000 +0800
+++ vision-0.22.0/cmake/TorchVisionConfig.cmake.in	2025-04-24 10:56:29.342314833 +0800
@@ -31,7 +31,9 @@ if(@WITH_CUDA@)
   target_compile_definitions(${PN}::${PN} INTERFACE WITH_CUDA)
 endif()
 
-find_package(Torch REQUIRED)
+if(NOT TARGET torch)
+  find_package(Torch REQUIRED)
+endif()
 target_link_libraries(${PN}::${PN} INTERFACE torch)
 
 if(@WITH_PNG@)
diff -Nuarp vision-0.22.0.orig/CMakeLists.txt vision-0.22.0/CMakeLists.txt
--- vision-0.22.0.orig/CMakeLists.txt	2025-04-14 16:19:44.000000000 +0800
+++ vision-0.22.0/CMakeLists.txt	2025-04-24 10:56:29.342457089 +0800
@@ -27,6 +27,7 @@ if(WITH_MPS)
 endif()
 
 find_package(Torch REQUIRED)
+include_directories(${TORCH_INCLUDE_DIRS})
 
 if (WITH_PNG)
     add_definitions(-DPNG_FOUND)
@@ -90,8 +91,8 @@ include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 
 set(TVCPP torchvision/csrc)
-list(APPEND ALLOW_LISTED ${TVCPP} ${TVCPP}/io/image ${TVCPP}/io/image/cpu ${TVCPP}/io/image/cpu/giflib ${TVCPP}/models ${TVCPP}/ops
-  ${TVCPP}/ops/autograd ${TVCPP}/ops/cpu ${TVCPP}/io/image/cuda)
+list(APPEND ALLOW_LISTED ${TVCPP} ${TVCPP}/ops
+  ${TVCPP}/ops/autograd ${TVCPP}/ops/cpu)
 if(WITH_CUDA)
     list(APPEND ALLOW_LISTED ${TVCPP}/ops/cuda ${TVCPP}/ops/autocast)
 endif()
@@ -103,7 +104,7 @@ FOREACH(DIR ${ALLOW_LISTED})
     file(GLOB ALL_SOURCES ${ALL_SOURCES} ${DIR}/*.*)
 ENDFOREACH()
 
-add_library(${PROJECT_NAME} SHARED ${ALL_SOURCES})
+add_library(${PROJECT_NAME} STATIC ${ALL_SOURCES})
 target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIBRARIES})
 
 if(WITH_MPS)
