diff -Nuarp vision-0.16.2.orig/torchvision/csrc/macros.h vision-0.16.2/torchvision/csrc/macros.h
--- vision-0.16.2.orig/torchvision/csrc/macros.h	2023-11-23 08:22:14.000000000 +0800
+++ vision-0.16.2/torchvision/csrc/macros.h	2023-12-15 15:38:00.655634874 +0800
@@ -1,14 +1,6 @@
 #pragma once
 
-#if defined(_WIN32) && !defined(TORCHVISION_BUILD_STATIC_LIBS)
-#if defined(torchvision_EXPORTS)
-#define VISION_API __declspec(dllexport)
-#else
-#define VISION_API __declspec(dllimport)
-#endif
-#else
 #define VISION_API
-#endif
 
 #if (defined __cpp_inline_variables) || __cplusplus >= 201703L
 #define VISION_INLINE_VARIABLE inline
