diff -Nuarp vision-0.27.1.orig/torchvision/csrc/vision.h vision-0.27.1/torchvision/csrc/vision.h
--- vision-0.27.1.orig/torchvision/csrc/vision.h	2026-06-10 07:54:21.000000000 +0800
+++ vision-0.27.1/torchvision/csrc/vision.h	2026-07-03 15:18:20.433891483 +0800
@@ -5,8 +5,4 @@
 
 namespace vision {
 VISION_API int64_t cuda_version();
-
-namespace detail {
-extern "C" inline auto _register_ops = &cuda_version;
-} // namespace detail
 } // namespace vision
