diff -Nuarp vision-0.22.1.orig/torchvision/csrc/macros.h vision-0.22.1/torchvision/csrc/macros.h
--- vision-0.22.1.orig/torchvision/csrc/macros.h	2025-05-09 21:43:46.000000000 +0800
+++ vision-0.22.1/torchvision/csrc/macros.h	2025-07-24 16:54:19.143664902 +0800
@@ -1,11 +1,3 @@
 #pragma once
 
-#if defined(_WIN32) && !defined(TORCHVISION_BUILD_STATIC_LIBS)
-#if defined(torchvision_EXPORTS)
-#define VISION_API __declspec(dllexport)
-#else
-#define VISION_API __declspec(dllimport)
-#endif
-#else
 #define VISION_API
-#endif
