diff -Nuarp vision-0.22.0.orig/torchvision/csrc/macros.h vision-0.22.0/torchvision/csrc/macros.h
--- vision-0.22.0.orig/torchvision/csrc/macros.h	2025-04-14 16:19:44.000000000 +0800
+++ vision-0.22.0/torchvision/csrc/macros.h	2025-04-24 10:56:00.350352414 +0800
@@ -1,11 +1,3 @@
 #pragma once
 
-#if defined(_WIN32) && !defined(TORCHVISION_BUILD_STATIC_LIBS)
-#if defined(torchvision_EXPORTS)
-#define VISION_API __declspec(dllexport)
-#else
-#define VISION_API __declspec(dllimport)
-#endif
-#else
 #define VISION_API
-#endif
