From fb771ef5fedb76aabf00c6c6b113ae16d059ba2d Mon Sep 17 00:00:00 2001
From: nihui <shuizhuyuanluo@126.com>
Date: Sun, 13 Mar 2022 11:02:44 +0800
Subject: [PATCH] add python linking dependency when USE_PYTHON=on

---
 cmake/TorchVisionConfig.cmake.in | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/cmake/TorchVisionConfig.cmake.in b/cmake/TorchVisionConfig.cmake.in
index aa04284475..98f4f58d3c 100644
--- a/cmake/TorchVisionConfig.cmake.in
+++ b/cmake/TorchVisionConfig.cmake.in
@@ -28,14 +28,16 @@ include("${CMAKE_CURRENT_LIST_DIR}/${PN}Targets.cmake")
 if(NOT TARGET torch_library)
 find_package(Torch REQUIRED)
 endif()
-if (@USE_PYTHON@)
+if(@USE_PYTHON@)
   if(NOT TARGET Python3::Python)
     find_package(Python3 COMPONENTS Development)
   endif()
 endif()
 
-set_target_properties(TorchVision::TorchVision PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${${PN}_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "torch;Python3::Python" )
-
+set_target_properties(TorchVision::TorchVision PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${${PN}_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "torch")
+if(@USE_PYTHON@)
+  set_property(TARGET TorchVision::TorchVision APPEND PROPERTY INTERFACE_LINK_LIBRARIES "Python3::Python")
+endif()
 
 if(@WITH_CUDA@)
   target_compile_definitions(TorchVision::TorchVision INTERFACE WITH_CUDA)
