diff -Nuarp vision-0.22.1.orig/torchvision/csrc/vision.h vision-0.22.1/torchvision/csrc/vision.h
--- vision-0.22.1.orig/torchvision/csrc/vision.h	2025-05-09 21:43:46.000000000 +0800
+++ vision-0.22.1/torchvision/csrc/vision.h	2025-07-24 16:53:48.341507602 +0800
@@ -6,7 +6,4 @@
 namespace vision {
 VISION_API int64_t cuda_version();
 
-namespace detail {
-extern "C" inline auto _register_ops = &cuda_version;
-} // namespace detail
 } // namespace vision
