diff -Nuarp vision-0.23.0.orig/torchvision/csrc/vision.h vision-0.23.0/torchvision/csrc/vision.h
--- vision-0.23.0.orig/torchvision/csrc/vision.h	2025-07-18 05:23:54.000000000 +0800
+++ vision-0.23.0/torchvision/csrc/vision.h	2025-08-07 10:35:03.351949523 +0800
@@ -6,7 +6,4 @@
 namespace vision {
 VISION_API int64_t cuda_version();
 
-namespace detail {
-extern "C" inline auto _register_ops = &cuda_version;
-} // namespace detail
 } // namespace vision
