diff -Nuarp vision-0.16.0.orig/torchvision/csrc/vision.h vision-0.16.0/torchvision/csrc/vision.h
--- vision-0.16.0.orig/torchvision/csrc/vision.h	2023-09-25 21:45:29.000000000 +0800
+++ vision-0.16.0/torchvision/csrc/vision.h	2023-10-10 16:32:17.329264833 +0800
@@ -5,12 +5,4 @@
 
 namespace vision {
 VISION_API int64_t cuda_version();
-
-namespace detail {
-extern "C" VISION_INLINE_VARIABLE auto _register_ops = &cuda_version;
-#ifdef HINT_MSVC_LINKER_INCLUDE_SYMBOL
-#pragma comment(linker, "/include:_register_ops")
-#endif
-
-} // namespace detail
 } // namespace vision
