diff -Nuarp vision-0.21.0.orig/torchvision/csrc/macros.h vision-0.21.0/torchvision/csrc/macros.h
--- vision-0.21.0.orig/torchvision/csrc/macros.h	2025-01-29 09:12:10.000000000 +0800
+++ vision-0.21.0/torchvision/csrc/macros.h	2025-04-01 15:54:52.991683443 +0800
@@ -1,11 +1,3 @@
 #pragma once
 
-#if defined(_WIN32) && !defined(TORCHVISION_BUILD_STATIC_LIBS)
-#if defined(torchvision_EXPORTS)
-#define VISION_API __declspec(dllexport)
-#else
-#define VISION_API __declspec(dllimport)
-#endif
-#else
 #define VISION_API
-#endif
