diff -Nuarp vision-0.22.0.orig/torchvision/csrc/vision.h vision-0.22.0/torchvision/csrc/vision.h
--- vision-0.22.0.orig/torchvision/csrc/vision.h	2025-04-14 16:19:44.000000000 +0800
+++ vision-0.22.0/torchvision/csrc/vision.h	2025-04-24 10:55:11.257237692 +0800
@@ -6,7 +6,4 @@
 namespace vision {
 VISION_API int64_t cuda_version();
 
-namespace detail {
-extern "C" inline auto _register_ops = &cuda_version;
-} // namespace detail
 } // namespace vision
