diff -Nuarp vision-0.14.0/CMakeLists.txt vision-0.14.0-ops-only/CMakeLists.txt
--- vision-0.14.0/CMakeLists.txt	2022-10-11 12:22:58.000000000 +0800
+++ vision-0.14.0-ops-only/CMakeLists.txt	2022-11-05 21:35:01.471076907 +0800
@@ -79,8 +79,8 @@ include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 
 set(TVCPP torchvision/csrc)
-list(APPEND ALLOW_LISTED ${TVCPP} ${TVCPP}/io/image ${TVCPP}/io/image/cpu ${TVCPP}/models ${TVCPP}/ops
-  ${TVCPP}/ops/autograd ${TVCPP}/ops/cpu ${TVCPP}/io/image/cuda)
+list(APPEND ALLOW_LISTED ${TVCPP} ${TVCPP}/ops
+  ${TVCPP}/ops/autograd ${TVCPP}/ops/cpu)
 if(WITH_CUDA)
     list(APPEND ALLOW_LISTED ${TVCPP}/ops/cuda ${TVCPP}/ops/autocast)
 endif()
