diff -Nuarp vision-0.11.1/cmake/TorchVisionConfig.cmake.in vision-0.11.1.no-python/cmake/TorchVisionConfig.cmake.in
--- vision-0.11.1/cmake/TorchVisionConfig.cmake.in	2021-10-21 09:16:22.000000000 +0800
+++ vision-0.11.1.no-python/cmake/TorchVisionConfig.cmake.in	2022-04-28 16:34:33.810073013 +0800
@@ -28,11 +28,8 @@ include("${CMAKE_CURRENT_LIST_DIR}/${PN}
 if(NOT TARGET torch_library)
 find_package(Torch REQUIRED)
 endif()
-if(NOT TARGET Python3::Python)
-find_package(Python3 COMPONENTS Development)
-endif()
 
-set_target_properties(TorchVision::TorchVision PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${${PN}_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "torch;Python3::Python" )
+set_target_properties(TorchVision::TorchVision PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${${PN}_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "torch" )
 
 
 if(@WITH_CUDA@)
